File: Volt.client/src/app/core/models/app-role.ts
/** Mirrors Domain.Enums.AppRole — keep in sync with backend. */
export enum AppRole {
  Customer = 1,
  SuperAdmin = 2,
  Merchant = 3,
  Delivery = 4
}

export const AppRoleNames = {
  Customer: 'Customer',
  SuperAdmin: 'Super Admin',
  Merchant: 'Merchant',
  Delivery: 'Delivery'
} as const;

export function appRoleFromName(roleName: string | null | undefined): AppRole | null {
  switch (roleName) {
    case AppRoleNames.Customer:
      return AppRole.Customer;
    case AppRoleNames.SuperAdmin:
      return AppRole.SuperAdmin;
    case AppRoleNames.Merchant:
      return AppRole.Merchant;
    case AppRoleNames.Delivery:
      return AppRole.Delivery;
    default:
      return null;
  }
}

export function appRoleName(role: AppRole): string {
  switch (role) {
    case AppRole.Customer:
      return AppRoleNames.Customer;
    case AppRole.SuperAdmin:
      return AppRoleNames.SuperAdmin;
    case AppRole.Merchant:
      return AppRoleNames.Merchant;
    case AppRole.Delivery:
      return AppRoleNames.Delivery;
    default:
      return '';
  }
}

File: Domain/Enums/AppRole.cs
namespace Domain.Enums
{
    public enum AppRole
    {
        Customer = 1,
        SuperAdmin = 2,
        Merchant = 3,
        Delivery = 4
    }

    public static class AppRoleNames
    {
        public const string Customer = "Customer";
        public const string SuperAdmin = "Super Admin";
        public const string Merchant = "Merchant";
        public const string Delivery = "Delivery";

        public static readonly string[] All =
        [
            Customer,
            SuperAdmin,
            Merchant,
            Delivery
        ];

        public static string ToRoleName(AppRole role) => role switch
        {
            AppRole.Customer => Customer,
            AppRole.SuperAdmin => SuperAdmin,
            AppRole.Merchant => Merchant,
            AppRole.Delivery => Delivery,
            _ => throw new ArgumentOutOfRangeException(nameof(role), role, "Unknown application role")
        };

        public static bool TryFromInt(int value, out AppRole role)
        {
            if (Enum.IsDefined(typeof(AppRole), value))
            {
                role = (AppRole)value;
                return true;
            }

            role = default;
            return false;
        }

        public static bool IsPublicRegistrationAllowed(AppRole role) =>
            role is AppRole.Customer or AppRole.Merchant or AppRole.Delivery;
    }
}
